ivy/functional/frontends/builtins/func_wrapper.py

```diff
@@ -0,0 +1,101 @@
+import functools
+from typing import Callable, Iterable
+from importlib import import_module
+
+import ivy
+
+
+def _to_ivy_array(x):
+    # if x is any frontend array including frontend list
+    # then extract the wrapped ivy array
+    if hasattr(x, "ivy_array"):
+        return x.ivy_array
+
+    # convert native arrays and lists to ivy arrays
+    elif isinstance(x, ivy.NativeArray):
+        return ivy.array(x)
+
+    return x
+
+
+def _infer_return_array(x: Iterable) -> Callable:
+    # get module path
+    module_path = x.__class__.__module__.split(".")
+
+    # if function's input is a scalar, list, or tuple
+    # convert to current backend's frontend array
+    if "builtins" in module_path:
+        cur_backend = ivy.current_backend_str()
+        module_str = cur_backend if len(cur_backend) != 0 else "numpy"
```

Comment:
Regardless of everything else, this would not be correct because we want to mimic the original function exactly and builtin functions never return an "array" asfaik.